OpenHardwareMonitor/GUI/StartupManager.cs

```diff
@@ -238,11 +234,23 @@ private void DeleteSchedulerTask()
             try
             {
                 // Previous versions might have registered under this name
-                root.DeleteTask("OpenHardwareMonitorAutoStart", 0);
+                root.DeleteTask("OpenHardwareMonitorAutoStart", 0); 
             }
             catch (IOException)
             {
             }
+
```

Comment:
Thanks for cleaning this up. While it's not a new problem, these "catch and ignore" handlers are very ugly. Can you add exception logging for the error cases, so that the log file contains information when any of this fails? You can get a logger by calling `Logging.GetLogger("some name")`.

Reply:
I didn't like that exception handling either, but I opted not to change it.

I've tried to implement logger support the same way as [Hardware/HDD/WindowsSmart.cs](https://github.com/hexagon-oss/openhardwaremonitor/blob/c60ac4bd1af591f8332ea035b2584ce46d66ce76/OpenHardwareMonitorLib/Hardware/HDD/WindowsSmart.cs#L225) in my commit 6231c0dcc8092ac34a6b9245a39402ed9e3b35a7 .

Please note, that I didn't wrote a single line of C#/.NET 6 code before these commits, and I have zero clue about how this logger library works. 
However, I wasn't able to crash the app by adding\removing tasks, closing on re-opening app, even after by setting ` taskCount = -5 ` to bypass the check. And there is an output in VStudio output window about these errors:
```
2023-08-15 04:45:29.9545|638276895299545874|1|Error|OpenHardwareMonitor.GUI.StartupManager: Папка не пуста. (0x80070091) Папка не пуста. (0x80070091)
2023-08-15 04:45:29.9545|638276895299545874|1|Error|OpenHardwareMonitor.GUI.StartupManager: Не удается найти указанный файл. (0x80070002) Не удается найти указанный файл. (0x80070002)
```
(Folder isn't empty and the file not found, respectively). I ain't sure if there was an output before, tho.

Reply:
Hey, that's very good for someone new to the .NET world. It's really almost perfect now, except that you should move the `this.GetCurrentClassLogger()` line to the constructor of the class, not repeat it in every exception.

Reply:
I've moved it to class constructor, please review the code